core/move.js

```diff
@@ -0,0 +1,47 @@
+/* @flow */
+
+const _ = require('lodash')
+
+/*::
+import type { Metadata } from './metadata'
+*/
+
+// Export so the following possible is possible:
+//
+//    const move = require('.../move')
+//    move(src, dst)
+//    move.child(src, dst)
+//
+module.exports = move
+move.child = child
+
+// Modify the given src/dst docs so they can be merged then moved accordingly
+// during sync.
+function move (src /*: Metadata */, dst /*: Metadata */) {
+  // moveTo is used for comparison. It's safer to take _id
+  // than path for this case, as explained in doc/developer/design.md
+  src.moveTo = dst._id
+  src._deleted = true
+
+  // Make sure newly moved docs have their fill of sync attempts
+  delete src.errors
+  delete dst.errors
+
+  // TODO: Find out wether or not it would make sense to also delete the
+  // trashed property on the source, or explain why it doesn't.
+  delete dst.trashed
+
+  dst.moveFrom = _.omit(src, 'moveFrom')
```

Comment:
`moveTo` should only be set on deleted docs, so it shouldn't be set anyway.
Maybe we could omit `errors` too.

@aenario The more I think about it, the more I believe we should actually remove any non-useful property to prevent confusion. And introduce some corresponding `MoveSource` type. We're mostly need `path`, `incompatibilities`. For some reason we're also using `remove._id` and `remote._rev` in `side.move*()` but I assume we could set those on the destination doc and use them from there... This is for a future PR though.